integration-tests/collectors/README.md

```diff
@@ -0,0 +1,84 @@
+# Collector test
+## Setup
+### Dependencies
+* GitHub repo: https://github.com/scoheb/e2e-base
+* GitHub personal access token (classic) for above repo with **admin:repo_hook**, **delete_repo**, **repo** scopes.
+* The password to the vault files. (Contact a member of the Release team should you want to run this
+  test suite.)
+* Access to the target cluster and tenant and managed namespaces
+  * This test uses stg-rh01 and the dev-release-team-tenant and managed-release-team-tenant namespaces.
+* At least one JIRA issue that can be found via this query:
+  * `'project = "Konflux Release"  AND summary ~ "test issue for collector e2e testing"'`
+### Required Environment Variables
+- GITHUB_TOKEN
+  - The GitHub personal access token needed for repo operations
+  - The repo in question can be located in [test.env](test.env)
+- VAULT_PASSWORD_FILE
+  - This is the path to a file that contains the ansible vault
+    password needed to decrypt the secrets needed for testing.
+- RELEASE_CATALOG_GIT_URL
+  - The release service catalog URL to use in the RPA
+  - This is provided when testing PRs
+- RELEASE_CATALOG_GIT_REVISION
+  - The release service catalog revision to use in the RPA
+  - This is provided when testing PRs
+### Optional Environment Variables
+- KUBECONFIG
+  - The KUBECONFIG file to used to login to the target cluster
+  - This is provided when testing PRs 
+### Test Properties
+#### [test.env](test.env)
+- This file contains resource names and configuration values needed for testing.
+- Since this test requires internal services, the tenant and managed namespaces
+  should remain as-is.
+### Test Functions
+#### [lib/test-functions.sh](../lib/test-functions.sh)
+- This file contains re-usable functions for tests
+### Secrets
+- Secrets needed for testing are stored in ansible vault files.
+  - [vault/collector-managed-secrets.yaml](vault/collector-managed-secrets.yaml)
+  - [vault/collector-tenant-secrets.yaml](vault/collector-tenant-secrets.yaml)
+- Most secrets required are contained in the files above.
+- Some tests have their secret name hardcoded and therefore must exist prior to running this test:
+  - konflux-advisory-jira-secret
+### Running the test
+
+The test has 2 modes:
+* With a CVE included and involved. (default mode)
+
+```shell
+run-test.sh
+```
+ 
+* Without a CVE involved.
+
+```shell
+run-test.sh --no-cve
+```
+
+### Debugging
+
+There is a `--skip-cleanup` option to the script in the event that you want to examine the resources
+after a test has ended.
+
+Note: you can use these 2 scripts to clean up when you are completed debugging:
+* [utils/cleanup-resources.sh](utils/cleanup-resources.sh)
```

Comment:
I didn't see where cleanup-resources.sh and delete-branches.sh are called in the scripts. If the test enters CI, it will be ran by each PR. It's best to delete test data(CRs and resources) after each run. Seems cleanup-resources.sh doesn't support parallel jobs.

Reply:
> cleanup-resources.sh and delete-branches.sh are only for periodic cleanup if required. In other words...only in emergency situations.

We need to clean up it to leave each CI job a clean environment.  Otherwise, it will affect the stability of the CI.
I  think it's better to setup the clean up mechanism before it's used.


> The run-test.sh has a trap that calls resource and branch cleanup functions that each run.

If another test ( triggered by another PR ) is running at the same time,  it will be affected.

Reply:
Have you reviewed the whole PR? 

Each resource has a unique name. There is no possibility for collision.

And since each run uses a unique resource naming scheme, it is not possible to cleanup before hand since the resource will never exist.

Reply:
If two PRs trigger two test runs concurrently, and one of them encounters an issue that invokes the resource and branch cleanup functions, these functions will clean up all collector-related resources and the branch. The other test run will be affected, right?

Reply:
Wrong.

Please see the trap function. It only cleans up the unique resources created by the test itself.

Multiple PRs can run at the same time without issue.

Reply:
Yes. You are right. I didn't notice you have two places to cleanup CRs.